tests/test_lowlevel.py

```diff
@@ -204,3 +203,26 @@ def test_use_proxy_from_environment(httpbin, var, scheme):
 
         # it had actual content (not checking for SOCKS protocol for now)
         assert len(fake_proxy.handler_results[0]) > 0
+
+
+def test_redirect_rfc1808_to_non_ascii_location():
+    def redirect_resp_handler(sock):
+        consume_socket_content(sock, timeout=0.5)
+        sock.send(
+            b'HTTP/1.1 301 Moved Permanently\r\n'
+            b'Location: //%s:%d/?\xc5\xa0\r\n'
+            b'\r\n' % (host.encode(), port)
+        )
+        consume_socket_content(sock, timeout=0.5)
+        sock.send(b'HTTP/1.1 200 OK\r\n\r\n')
+
+    close_server = threading.Event()
+    server = Server(redirect_resp_handler, wait_to_close_event=close_server)
+
+    with server as (host, port):
+        url = u'http://{0}:{1}'.format(host, port)
+        r = requests.head(url=url, allow_redirects=True)
+        assert r.status_code == 200
+        assert len(r.history) == 1
+        assert r.history[0].status_code == 301
```

Comment:
Hey @nateprewitt thanks for the quick review!

I would be glad to make it happen but the issue is that current decoding into native string of headers makes Location header on python2 encoded string and on python3 decoded string. so this test won't be consistent across python version.

before this PR we get for the same test the following redirected URL:
python3: http://localhost:58792/?%C3%85%C2%A0
python2: UnicodeDecodeError (due to implicit decoding into unicode on string formatting)

this PR only fixes the python2 UnicodeDecodeError (python3 behaves the same).

there is an encoding issue #3888 which leads to inconsistency between python versions as each python version handles different type of strings.

I agree that this PR is far from solving #3888 (which I'll be happy to discuss about possible solutions).

My ideas are (just note a few, regardless of order):

1. keep this PR as is
2. add xfail for this test
3. test only `the resolve_redirects()` function
4. discuss possible for solution for #3888 that I might implement in the same PR
5. leave all as is and ignore this PR

of course I am open to any suggestions :smile: